Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -409,6 +410,11 @@ void AppendPipeline::appsinkNewSample(const Track& track, GRefPtr<GstSample>&& s
         mediaSample->extendToTheBeginning();
     }
 
+    if (track.streamType == StreamType::Text) {
+        const auto textTrack = static_cast<InbandTextTrackPrivateGStreamer*>(track.webKitTrack.get());
+        textTrack->handleSample(GRefPtr(mediaSample->sample()));
+    }
+
     m_sourceBufferPrivate.didReceiveSample(mediaSample.get());
```

Comment:
We must not call `didReceiveSample()` if we don't have valid timestamps, I checked:

If we call `didReceiveSample()` with a sample that has no timestamps, it will end up in `SourceBufferPrivate::processMediaSample()`, and inside there `sample->presentationTime()` is called and arithmetic is done on it. That's going to lead to problems if the presentationTime is set as invalid.

For video tracks we avoid this problem with the early return from before:

```
   if (!GST_BUFFER_PTS_IS_VALID(buffer) && track.streamType != Text) {
       // [...]
       return;
```

However, since we don't do the early return for text tracks in order to handle the WebVTT headers, we need to guard against invalid PTS again here.